engine/src/main/java/org/hibernate/search/query/dsl/sort/SortAdditionalSortFieldContext.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.query.dsl.sort;
+
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.SortField;
+
+/**
+ * @author Emmanuel Bernard emmanuel@hibernate.org
+ */
+public interface SortAdditionalSortFieldContext {
+
+	// navigation to next sort definition
+
+	/**
+	 * Order elements by a specific field
+	 */
+	SortFieldContext andByField(String field);
+
+	/**
+	 * Order elements by their relevance score in decreasing order by default.
+	 */
+	//TODO Is byScore terminal or can it be followed by other sorts?
+	SortOrderTermination andByScore();
+
+	/**
+	 * Order elements by their internal index order.
+	 * This is the fastest sorting.
+	 */
+	SortOrderTermination andByIndexOrder();
```

Comment:
Similarly to _andByScore_, I'd remove  this option too as it makes even less sense to use in practice to mix another sorting with the "by index" sort.

I think "by index" is used exclusively for fastest speed when sorting doesn't matter at all (in which case one wouldn't combine it with another sort), or if the index was specifically constructed to be laid out in a specific sort order but we don't give such control.